test/util.test.js

```diff
@@ -208,9 +208,62 @@ describe('mergeSettings', function() {
 
     should.deepEqual(dst.acls, expected.acls, 'Merged settings should match the expectation');
   });
+
+  it('should merge relations correctly', function() {
+    var src = {
+      relations: {
+        user: {
+          type: 'belongsTo',
+          idName: 'id',
+          polymorphic: {
+            idType: 'string',
+            foreignKey: 'userId',
+            discriminator: 'principalType',
+          },
+        },
+      },
+    };
+    var tgt = {
+      relations: {
+        user: {
+          type: 'belongsTo',
+          model: 'User',
+          foreignKey: 'userId',
+        },
+        someRelation: {
+          type: 'hasMany',
+          model: 'someModel',
+          foreignKey: 'modelId',
+        },
+      },
+    };
+
+    var dst = mergeSettings(tgt, src);
```

Comment:
+1, I would personally also switch the order in which the variables are defined - start with the parent, then define a child. I think it would be helpful to add comments explaining what these settings are mimicking (LoopBack's built-in AccessToken model, a custom model extending the built-in AccessToken model).